web/profiles/custom/yalesites_profile/modules/custom/ys_views_basic/src/Plugin/views/style/ViewsBasicDynamicStyle.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Drupal\ys_views_basic\Plugin\views\style;
+
+use Drupal\views\Plugin\views\style\StylePluginBase;
+
+/**
+ * Unformatted style plugin to render rows with dynamic view mode.
+ *
+ * Row are rendered one after another with no decorations.
+ *
+ * @ingroup views_style_plugins
+ *
+ * @ViewsStyle(
+ *   id = "ys_views_basic_dynamic_style",
+ *   title = @Translation("Views Basic Dynamic Style"),
+ *   help = @Translation("Displays rows one after another."),
+ *   theme = "views_view_unformatted",
+ *   display_types = {"normal"}
+ * )
+ */
+class ViewsBasicDynamicStyle extends StylePluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $usesRowPlugin = TRUE;
+
+  /**
+   * Does the style plugin support custom css class for the rows.
+   *
+   * @var bool
+   */
+  protected $usesRowClass = TRUE;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preRender($result) {
+    if (!empty($this->view->rowPlugin)) {
+
+      // Gets passed view mode from ViewsBasicDefaultFormatter and sets per row.
+      if (isset($this->view->args[3])) {
+        $viewMode = $this->view->args[3];
+        $validViewModes = \Drupal::service('entity_display.repository')->getViewModeOptions('node');
```

Comment:
> Almost there just a minor improvement over the code that needs to be done and also I have a question about the first page Element and Last page element , is it ok for them to keep the number instead of the words First page or last page?
> <img alt="Screenshot 2023-02-23 at 3 43 32 PM" width="764" src="https://user-images.githubusercontent.com/2217827/221037667-9c8fe65f-3441-4c54-b0e7-8f77ab965d3a.png">
> I think that having that as it is right could be kind of confusing for some users , but I'm just a dev not a UX haha , let me know what you think about it @marcb4k

@joetower - Would like to discuss the pager styling with you. Is this normal? I was just using the "full" pager style and I checked against the pager styling in Storybook here (https://yalesites-org.github.io/component-library-twig/?path=/story/molecules-pager--first-and-last) and it looks like it is correct - i.e. the "first page" is labeled as "1", but also there would be a "page 1" link that is also labeled "1" if you are on, say, page 2.

@rigoucr - I'll fix the dependency injection thing and let you know when that's ready for re-review.

Reply:
@rigoucr, refactored for dependency injection. Good catch, thank you!

Reply:
@rigoucr - could you take a quick look at the last update with the dependency injection? Then we can merge. Thank you!